pkg/sendconfig/common_workflows.go

```diff
@@ -0,0 +1,58 @@
+package sendconfig
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kong/kubernetes-ingress-controller/pkg/deckgen"
+	"github.com/kong/kubernetes-ingress-controller/pkg/parser"
+	"github.com/kong/kubernetes-ingress-controller/pkg/store"
+)
+
+// UpdateKongAdminSimple is a helper function for the most common usage of PerformUpdate() with only minimal
+// upfront configuration required. This function is specialized and highly opinionated.
+//
+// If you're implementation needs to expand on the configuration and usage of the following inner components:
+//
+//   - store.Storer
+//   - kongstate.Kong
+//   - deckgen.ToDeckContent()
+//   - sendconfig.PerformUpdate()
+//
+// Or any other encapsulated components this function makes all of that opaque to the caller.
+// Treat this function as a very specific "workflow" to update the Kong Admin API,
+// and use it as a reference to implement the workflow you need.
+func UpdateKongAdminSimple(ctx context.Context,
+	lastConfigSHA []byte,
+	cache *store.CacheStores,
+	ingressClassName string,
+	deprecatedLogger logrus.FieldLogger,
+	kongConfig Kong,
+	enableReverseSync bool,
+) ([]byte, error) {
```

Comment:
How is it possible that codecov reports this as untested? Shouldn't the integration tests be included in code coverage? Doesn't this function get run as a part thereof?

Reply:
There's a bug that was previously reported for this problem #1310